libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -948,8 +957,22 @@ void AC_PosControl::update_vel_controller_xyz()
 
 /// calc_leash_length - calculates the horizontal leash length given a maximum speed, acceleration
 ///     should be called whenever the speed, acceleration or position kP is modified
+///     this also applies the PSC_MAX_LEASH parameter if it is present
 void AC_PosControl::calc_leash_length_xy()
 {
+    // Check if PSC_MAX_LEASH (_max_leash_length) agrees with _current_max_leash_length
+    if (!is_equal(_current_max_leash_length, static_cast<float>(_max_leash_length))) {
```

Comment:
```suggestion
    if (!is_equal(_current_max_leash_length, _max_leash_length.get())) {
```

(we have devs who are averse to static casts....)

Reply:
Haha. That is understandable. Thanks for the tips!